From 197d8bb3045f28468232f8943ca231184371f217 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 30 Aug 2016 13:41:26 +0200
Subject: [PATCH] Honor HTTPS_CERT_FILE and HTTPS_KEY_FILE for IO::Socket:SSL
 class
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Changing default SSL implementation from Net::SSL to IO::Socket::SSL
broke Net::SSL way for specifying client key and certificate via
enviroment variables.

This patch translates these two environment variables into
IO::Socket::SSL constructor arguments to provide compatibility with
Net::SSL.

<https://bugzilla.redhat.com/show_bug.cgi?id=1365490>

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 lib/LWP/UserAgent.pm | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/lib/LWP/UserAgent.pm b/lib/LWP/UserAgent.pm
index a884164..6d526a4 100644
--- a/lib/LWP/UserAgent.pm
+++ b/lib/LWP/UserAgent.pm
@@ -66,6 +66,16 @@ sub new
             $ssl_opts->{SSL_ca_path} = $ca_path;
         }
     }
+    unless (exists $ssl_opts->{SSL_cert_file}) {
+        if (my $ca_path = $ENV{HTTPS_CERT_FILE}) {
+            $ssl_opts->{SSL_cert_file} = $ca_path;
+        }
+    }
+    unless (exists $ssl_opts->{SSL_key_file}) {
+        if (my $ca_path = $ENV{HTTPS_KEY_FILE}) {
+            $ssl_opts->{SSL_key_file} = $ca_path;
+        }
+    }
     my $use_eval = delete $cnf{use_eval};
     $use_eval = 1 unless defined $use_eval;
     my $parse_head = delete $cnf{parse_head};
-- 
2.5.5

